### PR TITLE
Default value/not working update default grpc value

### PR DESCRIPTION
**Pull request by AMontagu**

### CODE REVIEW

File: django_socio_grpc/proto_serializers.py [hunk 1]
```diff
@@ -65,23 +68,65 @@ def populate_dict_with_none_if_not_required(self, data_dict, message=None):
                 code="missing_partial_message_attribute",
             )
 
+        is_update_process = bool(data_dict.get(self.Meta.model._meta.pk.name, ""))
```
Comment (legau) [inline django_socio_grpc/proto_serializers.py:71]: `self.Meta.model._meta.pk.name in data_dict`

File: django_socio_grpc/proto_serializers.py [hunk 2]
```diff
@@ -65,23 +68,65 @@ def populate_dict_with_none_if_not_required(self, data_dict, message=None):
...
-            if field.allow_null or field.default in [None, empty] and field.required is True:
+
+            if self.partial and field.field_name in data_dict.get(
+                PARTIAL_UPDATE_FIELD_NAME, {}
+            ):
+                if field.allow_null:
+                    data_dict[field.field_name] = None
+                    continue
+                if field.default not in [None, empty]:
+                    # TODO - AM - 12/01/2024 - is field.default a possible callable here ?
```
Comment (legau) [inline django_socio_grpc/proto_serializers.py:91]: should be done

File: django_socio_grpc/proto_serializers.py [hunk 3]
```diff
@@ -65,23 +68,65 @@ def populate_dict_with_none_if_not_required(self, data_dict, message=None):
...
+                    data_dict[field.field_name] = field.default
+                    continue
+
+                data_dict[field.field_name] = message.DESCRIPTOR.fields_by_name[
```
Comment (legau) [inline django_socio_grpc/proto_serializers.py:95]: this should throw imo, that means you explicitely put null into a non nullable field

File: django_socio_grpc/proto_serializers.py [hunk 4]
```diff
@@ -65,23 +68,65 @@ def populate_dict_with_none_if_not_required(self, data_dict, message=None):
...
+                    field.field_name
+                ].default_value
+
+            # INFO - AM - 12/01/2024 - if field name is not in data_dict but is required that mean that it's a default value deleted when transforming proto to dict
```
Comment (legau) [inline django_socio_grpc/proto_serializers.py:99]: This case doesn't exist, a default value is not deleted when using including_default_value_fields

File: django_socio_grpc/proto_serializers.py [hunk 5]
```diff
@@ -65,23 +68,65 @@ def populate_dict_with_none_if_not_required(self, data_dict, message=None):
...
+                    deferred_attribute = getattr(self.Meta.model, field.field_name)
+                    if deferred_attribute.field.default != NOT_PROVIDED:
+                        data_dict[field.field_name] = deferred_attribute.field.default
+                        continue
+
                 data_dict[field.field_name] = None
+                continue
+
+            # elif field.required and field.field_name in message.DESCRIPTOR.fields_by_name:
+            #     data_dict[field.field_name] = message.DESCRIPTOR.fields_by_name[field.field_name].default_value
```
Comment (legau) [inline django_socio_grpc/proto_serializers.py:129]: -

File: django_socio_grpc/proto_serializers.py (hunk 4 above)
Comment (AMontagu) [reply django_socio_grpc/proto_serializers.py:99]: including_default_value_fields is not used anymore

File: django_socio_grpc/proto_serializers.py (hunk 3 above)
Comment (AMontagu) [reply django_socio_grpc/proto_serializers.py:95]: may worth the shot to delete it and see which test fail

File: django_socio_grpc/proto_serializers.py (hunk 2 above)
Comment (AMontagu) [reply django_socio_grpc/proto_serializers.py:91]: indeed same for model default

File: django_socio_grpc/proto_serializers.py (hunk 5 above)
Comment (AMontagu) [reply django_socio_grpc/proto_serializers.py:129]: ? but yes to delete

**Pull request merged**